manifests/vhost/custom.pp

```diff
@@ -0,0 +1,38 @@
+# See README.md for usage information
+define apache::vhost::custom(
+  $content,
+  $ensure = 'present',
+  $priority = '25',
+) {
+  include ::apache
+
+  ## Apache include does not always work with spaces in the filename
+  $filename = regsubst($name, ' ', '_', 'G')
+
+  ::apache::custom_config { $filename:
+    ensure   => $ensure,
+    confdir  => $::apache::vhost_dir,
+    content  => $content,
+    priority => $priority,
+  }
+
+  # NOTE(pabelanger): This code is duplicated in ::apache::vhost and needs to
+  # converted into something generic.
+  if $::apache::vhost_enable_dir {
+    $vhost_symlink_ensure = $ensure ? {
+      present => link,
+      default => $ensure,
+    }
+
+    file { "${priority}-${filename}.conf symlink":
+      ensure  => $vhost_symlink_ensure,
+      path    => "${::apache::vhost_enable_dir}/${priority}-${filename}.conf",
+      target  => "${::apache::vhost_dir}/${priority}-${filename}.conf",
+      owner   => 'root',
+      group   => $::apache::params::root_group,
```

Comment:
Good point.  After some reading, it was initially included as root/root and then expanded out to support different root_groups based on distros in this commit https://github.com/puppetlabs/puppetlabs-apache/commit/b87ce2d598e729729a5ef4f7b5c88230f9ba7b07
So it looks like vhost conf files are always owned by root as far as the apache module is concerned?  I have no idea if that's correct as far as Apache itself is concerned or not but it seems to be working so far.  So I guess par for course is the correct method to follow on this update.